armada/api/middleware.py

```diff
@@ -11,17 +11,43 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-#
+
+# from keystoneclient import session
+# from keystoneclient.auth.identity import v3
+# from keystoneclient.v3 import client, roles, users
+
+from armada.conf import default
+
+# Required Oslo configuration setup
+default.register_opts()
+
+from keystoneclient.auth.identity import v3
+from keystoneclient.v3 import client
+
+from oslo_config import cfg
+from oslo_log import log as logging
 
 import falcon
 
+LOG = logging.getLogger(__name__)
+CONF = cfg.CONF
+DOMAIN = "armada"
+
+logging.setup(CONF, DOMAIN)
+
 class AuthMiddleware(object):
+
+    def __init__(self):
+        auth = v3.Password(auth_url=CONF.auth_url,
+                            username=CONF.username,
+                            password=CONF.password,
+                            project_name=CONF.project_name,
+                            user_domain_name=CONF.user_domain_name,
+                            project_domain_name=CONF.project_domain_name)
+
+        auth_session = session.Session(auth=auth)
+        self._client = client.Client(session=session)
+
     def process_request(self, req, resp):
-        token = req.get_header('X-Auth-Token')
-        self.validate_token(token)
-
-    def validate_token(self, token):
-        if token is None:
-            raise falcon.HTTPUnauthorized('X-Auth token invalid.')
-        if token != 'armada':
-            raise falcon.HTTPUnauthorized('X-Auth token invalid.')
+        token = req.get_header('X-Auth-Token', required=True)
+        return self._client.tokens.validate(token=token)
```

Comment:
I don't think the return value of the middleware process_* methods have any impact. This middleware should use the Keystone token to get the authenticated username and roles and add them to a context object. The downstream handler for the request can then use the context to authorize the roles for the actions being requested.